### PR TITLE
Bolt v2.5.2 Hotfixes

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-1483
http://vjira2:8080/browse/BDS-1484
http://vjira2:8080/browse/BDS-1485

## Summary
Fixes a handful of build and DevOps-related bugs occurring when running a build outside of the Bolt monorepo. Also adds a new Drupal Lab build step to the default Travis CI jobs to now test installing and building outside of the monorepo-installed context. 

## Details
- fixes a broken package.json path (used for determining the current version of Bolt being run) when build tools are run outside of the monorepo
- fixed a separate NPM package version mismatch issue in `@bolt/core`'s polyfills -- this fix updates the filenames to a handful of polyfills used internally
- updated the `@bolt/build-tools`'s Webpack build to automatically add the `@bolt/core/styles/index.scss` file to the default global entrypoint. Addresses a strange behavior observed where the exported JSON data for Sass files in `@bolt/core` weren't being generated as expected when compiling outside of the Bolt monorepo.
- Adds a new Drupal Lab Travis job to now start to check for these unexpected build-related issues when running the codebase downstream

## How to test
- Confirm the Travis build passes
- Confirm the new Drupal Lab Travis CI job (which adds these 3 package patches) builds without any errors.

### CODE REVIEW

**Comment by remydenton** [comment]
As a hotfix especially (though it doesn't hurt in general), I'd like to keep this as streamlined as possible.  Basically, these are the core three fixes we actually need:

c5051fb
f0eacc4
a2faa221

Beyond that, changes should tread as lightly as possible.  Tests are good, but if we all of a sudden get into the weeds trying to fix other stuff so that tests work, I worry that something will slip through the cracks.  A few specific suggestions 

## Git hygiene
Combine these two commits when you're writing them.  It's as simple as adding the `--amend` flag when you commit the second one:
2fc36cd8
38e85adf

Combine these as well.  You might need to do an interactive rebase.  The first few times you do that will take awhile, but it'll soon become quick and is worthwhile to reduce cognitive overhead both for reviewers and yourself:
bafea06
4b371fd

## Postpone optional fixes
Is 35ea6095 necessary to fix a particular problem/bug?  If so, I'd like to see a better description of what it's fixing.  If not, I don't think it's appropriate for a hotfix.

## Re: Drupal npm patches
I was thinking of the changes in 82bbd87 as something that would only ever happen in a codebase like pega.com in order to temporarily move past a particular issue and continue testing.  Seems like when you're working in Bolt, the fix to the Bolt codebase directly should be what we're testing (otherwise, there's a chance the patch and actual changes could differ...).  Are we using these patches because there's no way to pull the other changes into Drupal lab until they're published?  If so, I'd recommend at least removing the changes to the package dependencies in bafea06.  A problem to solve later, ideally.

In summary: The fixes are correct, but let's minimize noise in hotfix PRs especially (and all PRs when reasonable).

**Comment by sghoweri** [comment]
> As a hotfix especially (though it doesn't hurt in general), I'd like to keep this as streamlined as possible.  Basically, these are the core three fixes we actually need:
>
> c5051fb
> f0eacc4
> a2faa221
> 
> Beyond that, changes should tread as lightly as possible.  Tests are good, but if we all of a sudden get into the weeds trying to fix other stuff so that tests work, I worry that something will slip through the cracks.  A few specific suggestions 

@remydenton in general I totally agree with you, _however_ I also want to make sure we're addressing the core root cause of the problem that required the hotfix in the first place (ie. have a test that installs and runs the build outside of the Lerna-installed monorepo to catch issues -- even if those aren't discovered until right after a full Bolt release is published).

**Comment by sghoweri** [comment]
> ## Git hygiene
> Combine these two commits when you're writing them.  It's as simple as adding the `--amend` flag when you commit the second one:
> 2fc36cd8
> 38e85adf
> 
> Combine these as well.  You might need to do an interactive rebase.  The first few times you do that will take awhile, but it'll soon become quick and is worthwhile to reduce cognitive overhead both for reviewers and yourself:
> bafea06
> 4b371fd

@remydenton thanks -- definitely something I'll keep working on!

Out of curiosity, given that most of these extra commits you're referencing were changes I found myself needing to make after committing + pushing (and I know rebasing is it's own separate can of worms once changes have been pushed), would you have handled this any differently?

The biggest thing I was focusing on here was the final resulting file changes (even if I had to try a thing or two to get there).

**Comment by sghoweri** [comment]
> ## Postpone optional fixes
> Is 35ea6095 necessary to fix a particular problem/bug?  If so, I'd like to see a better description of what it's fixing.  If not, I don't think it's appropriate for a hotfix.

@remydenton I'll double-check but I do believe that popped up (threw an error) when I was testing these changes out locally. I'll definitely confirm and nix these if that wasn't the case.

**Comment by remydenton** [comment]
Re: testing, I rescind that part-- you should do what you need to do to have proper tests.  I'd just like the distinction to be as clear as possible between what is the core fix and what is testing-related. 

Re: git hygiene, if you've already pushed you're correct, you technically shouldn't do amending/rebasing.  I didn't realize that was the case.  Even then, I myself would amend and force push if I had just pushed the work and hadn't created a PR yet.  It's a gray area though.

**Comment by sghoweri** [comment]
> ## Re: Drupal npm patches
> I was thinking of the changes in 82bbd87 as something that would only ever happen in a codebase like pega.com in order to temporarily move past a particular issue and continue testing.  Seems like when you're working in Bolt, the fix to the Bolt codebase directly should be what we're testing (otherwise, there's a chance the patch and actual changes could differ...).  Are we using these patches because there's no way to pull the other changes into Drupal lab until they're published?  If so, I'd recommend at least removing the changes to the package dependencies in bafea06.  A problem to solve later, ideally.

@remydenton totally agree and for any packages that are installed and managed by Lerna (ie. anything in the `packages` folder) this would 100% be the case, however:

1. The Travis config update + the corresponding temp patch in Drupal Lab are both things that fall outside of what would get published to NPM as part of the hotfix (would 100% agree with not including any absolutely non-essential changes if these were part of the packages that would be getting published to NPM)

2. More practically, because this test (to try and prevent these sorts of issues from reoccurring) specifically involved installing and building the code from outside the monorepo-installed context, this was the only temp solution I could come up with (other than not getting any tests in place) to add the right checks we need + not have our Travis build break in the process. 😉

**Comment by sghoweri** [comment]
> > ## Postpone optional fixes
> > Is [35ea609](https://github.com/bolt-design-system/bolt/commit/35ea6095951150fb465bd99728b51b39daed13a3) necessary to fix a particular problem/bug?  If so, I'd like to see a better description of what it's fixing.  If not, I don't think it's appropriate for a hotfix.
> 
> @remydenton I'll double-check but I do believe that popped up (threw an error) when I was testing these changes out locally. I'll definitely confirm and nix these if that wasn't the case.

@remydenton I can confirm that those changes don't appear to be needed (weird... I swear I was seeing a build error before... 😬) -- I've reversed that commit and pushed that one change up so this should be good for you to double-check!

**Comment by sghoweri** [comment]
Thanks @remydenton!
